### PR TITLE
fix: :bug: verify function actually throws error

### PR DESCRIPTION
**Pull request by lwjohnst86**
This function should have thrown an error but didn't (mainly because I didn't yet know how to make give a useful error message). Now it does!

### CODE REVIEW

File: R/verify-variables.R [hunk 1]
```diff
@@ -6,26 +6,31 @@
...
 #' @keywords internal
 #'
 #' @examples
 #' # TODO: Replace with simulated data.
-#' example_bef_data <- tibble::tibble(pnr = 1, koen = 1, foed_dato = 1)
-#' osdc:::verify_required_variables(example_bef_data, "bef")
-verify_required_variables <- function(data, register) {
+#' verify_required_variables(register_data$bef, "bef")
+#' verify_required_variables(register_data$bef, "lpr_adm")
+verify_required_variables <- function(data, register, call = rlang::caller_env()) {
```
Comment (lwjohnst86) [inline R/verify-variables.R:20]: The `call` argument and `rlang::caller_env()` allows the error message to occur in the function this is used in. That way, the log traceback message doesn't show this function, only the function where the error should be shown (e.g. `join_lpr2()`).

Comment (Aastedet) [reply R/verify-variables.R:20]: That's neat!

File: R/verify-variables.R [hunk 2]
```diff
@@ -6,26 +6,34 @@
...
+#' @return Either TRUE if the verification passes, or an error.
 #' @keywords internal
 #'
 #' @examples
 #' # TODO: Replace with simulated data.
-#' example_bef_data <- tibble::tibble(pnr = 1, koen = 1, foed_dato = 1)
-#' osdc:::verify_required_variables(example_bef_data, "bef")
-verify_required_variables <- function(data, register) {
+#' verify_required_variables(register_data$bef, "bef")
+#' verify_required_variables(register_data$bef, "lpr_adm")
```
Comment (Aastedet) [inline R/verify-variables.R:19]: Possible typo? should this be ...(register_data$lpr_adm, "lpr_adm")?

Comment (lwjohnst86) [inline R/verify-variables.R:19]:
```suggestion
#' verify_required_variables(register_data$lpr_adm, "lpr_adm")
```

**Review by signekb — APPROVED** [review]
This is very niceee 👍

**Pull request merged**